tests/lax_control_flow_test.py

```diff
@@ -96,6 +96,76 @@ def posify(matrix):
   jtu.ignore_warning, message=".*jit-of-pmap.*")
 
 
+# Simple optimization routine for testing custom_root
+def binary_search(func, x0, low=0.0, high=100.0):
+  del x0  # unused
+
+  def cond(state):
+    low, high = state
+    midpoint = 0.5 * (low + high)
+    return (low < midpoint) & (midpoint < high)
+
+  def body(state):
+    low, high = state
+    midpoint = 0.5 * (low + high)
+    update_upper = func(midpoint) > 0
+    low = jnp.where(update_upper, low, midpoint)
+    high = jnp.where(update_upper, midpoint, high)
+    return (low, high)
+
+  solution, _ = lax.while_loop(cond, body, (low, high))
+  return solution
+
+# Optimization routine for testing custom_root.
+def newton_raphson(func, x0):
+  tol = 1e-7
+  max_it = 20
+
+  fx0, dfx0 = func(x0), jax.jacobian(func)(x0)
+  initial_state = (0, x0, fx0, dfx0)  # (iteration, x, f(x), grad(f)(x))
+
+  def cond(state):
+    it, _, fx, _ = state
+    return (jnp.max(jnp.abs(fx)) > tol) & (it < max_it)
+
+  def body(state):
+    it, x, fx, dfx = state
+    step = linear_equation_solve(dfx, fx + 1e-5)
+    x_next = x - step
+    fx, dfx = func(x_next), jax.jacobian(func)(x_next)
+    return (it + 1, x_next, fx, dfx)
+
+  _, x, _, _ = lax.while_loop(cond, body, initial_state)
+
+  return x
+
+
+# Optimization routine solving linear equations, i.e., f(x) = A * x + b
+def vector_solve(f, y):
+  return jnp.linalg.solve(jax.jacobian(f)(y), y)
+
+
+def linear_equation_solve(A, b):
+  """Solve A*x = b for x.
+
+  Needed because linalg.solve does not work for scalar
+  equations.
+  """
+  # Are there better ways to do a switch on the dimension?
+  return lax.cond(
+    b.size == 1,
+    # The max() helps the JIT understand
+    # that we get the same output size for both
+    # branches.
+    lambda _: jnp.divide(b, A.max()).reshape(b.shape),
+    lambda _: jnp.linalg.solve(
+        A.reshape((-1,b.size)),
+        b.reshape((b.size, 1))
+      ).reshape(b.shape),
+    None,
+  )
```

Comment:
Yes -- you can just do a normal Python `if` here. `b.size` is available as static information, e.g.,
```python
  if b.size == 1:
    return b / A
  else:
    return jnp.linalg.solve(
        A.reshape((-1, b.size)), b.ravel()
    ).reshape(b.shape)
```